lib/rucio/transfertool/fts3_new.py

```diff
@@ -0,0 +1,637 @@
+from rucio.core.monitor import record_counter, record_timer
+from rucio.db.sqla.constants import FTSState
+from rucio.transfertool.transfertool import Transfertool
+
+logging.getLogger("requests").setLevel(logging.CRITICAL)
+disable_warnings()
+
+logging.basicConfig(stream=sys.stdout,
+                    level=getattr(logging,
+                                  config_get('common', 'loglevel',
+                                             raise_exception=False,
+                                             default='DEBUG').upper()),
+                    format='%(asctime)s\t%(process)d\t%(levelname)s\t%(message)s')
+
+__USERCERT = config_get('conveyor', 'usercert', False, None)
+__USE_DETERMINISTIC_ID = config_get_bool('conveyor', 'use_deterministic_id', False, False)
+
+REGION_SHORT = make_region().configure('dogpile.cache.memory',
+                                       expiration_time=1800)
+
+
+class FTS3Transfertool(Transfertool):
+    """
+    FTS3 implementation of a Rucio transfertool
+    """
+
+    # Public methods part of the common interface
+
+    def submit(self, files, job_params, timeout=None):
+        """
+        Submit transfers to FTS3 via JSON.
+
+        :param files:        List of dictionaries describing the file transfers.
+        :param job_params:   Dictionary containing key/value pairs, for all transfers.
+        :param timeout:      Timeout in seconds.
+        :returns:            FTS transfer identifier.
+        """
+
+        # FTS3 expects 'davs' as the scheme identifier instead of https
+        for file in files:
+            if not file['sources'] or file['sources'] == []:
+                raise Exception('No sources defined')
+
+            new_src_urls = []
+            new_dst_urls = []
+            for url in file['sources']:
+                if url.startswith('https'):
+                    new_src_urls.append(':'.join(['davs'] + url.split(':')[1:]))
+                else:
+                    new_src_urls.append(url)
+            for url in file['destinations']:
+                if url.startswith('https'):
+                    new_dst_urls.append(':'.join(['davs'] + url.split(':')[1:]))
+                else:
+                    new_dst_urls.append(url)
+
+            file['sources'] = new_src_urls
+            file['destinations'] = new_dst_urls
+
+        transfer_id = None
+        expected_transfer_id = None
+        if __USE_DETERMINISTIC_ID:
+            job_params = job_params.copy()
+            job_params["id_generator"] = "deterministic"
+            job_params["sid"] = files[0]['metadata']['request_id']
+            expected_transfer_id = self.__get_deterministic_id(job_params["sid"])
+            logging.debug("Submit bulk transfers in deterministic mode, sid %s, expected transfer id: %s" % (job_params["sid"], expected_transfer_id))
+
+        # bulk submission
+        params_dict = {'files': files, 'params': job_params}
+        params_str = json.dumps(params_dict)
+
+        r = None
```

Comment:
pylint don't like one char varaibles ;-)

Reply:
The fts3 code is the copy&paste of the current fts3 code with some minor changes. I am not planning to change this massively at the moment is it is working...